README.md

```diff
@@ -70,8 +75,59 @@ The following cmake options are available and can also be specified explicitly f
+```shell
+detray-build/bin/detray_generate_toy_detector --write_material --write_grids
+```
+All of the validation tools presented in the following can also be run as part of a corresponding [python script](https://github.com/acts-project/detray/tree/main/tests/tools/python) which takes the same arguments and will automatically create plots from the collected data. However, this requires Python 3, pandas, SciPy and NumPy, as well as Matplotlib to be available.
+
+The detector geometry can be visualized in svg format with the following command:
+```shell
+detray-build/bin/detray_detector_display \
+   --geometry_file  ./toy_detector/toy_detector_geometry.json
+```
+The tool can also display single volumes or surfaces, as well as the navigation grids and material maps (the corresponding json files need to loaded in this case). For an overview of all available options for the command-line tools add ```--help```.
+
+### Navigation Validation
+
+In order to validate that the navigation works correctly in a given detector geometry, run the detector validation tool. It will first perform a consistency check on the detector, followed by a "ray scan" of the detector. The scan result will be compared to a full straight-line navigation run for every ray. After that, the navigation in a constant magnetic field of 2T is being tested in a similar fashion, using parameterized helix trajectories and a Newton-Raphson/Bisection algorithm to generate the truth intersections. For example:
+```shell
+detray-build/bin/detray_detector_validation \
+    --geometry_file ./toy_detector/toy_detector_geometry.json \
+    --grid_file ./toy_detector/toy_detector_surface_grids.json \
+    --search_window 3 3 --n_tracks 100 --pT_range 0.5 100
+```
+In case of failures, this command will give a detailed debug output in the form of a log file, as well as an svg representation of the failed tracks. The grid file is optional, but will trigger the use of spacial grids as acceleration structures during the navigation run.
+
+### Material Validation
+
+This tool checks whether the navigator picks up the material correctly by comparing the material found during a ray scan with the material collected during navigation by a specialized actor:
+```shell
+detray-build/bin/detray_material_validation \
+    --geometry_file toy_detector/toy_detector_geometry.json \
+    --material_file toy_detector/toy_detector_homogeneous_material.json \
+    --phi_steps 100 --eta_steps 100 --eta_range -4 4
+```
+Note: The correct material file must be loaded in addition to the geometry file!
+
+
+## Benchmarks
+
+A number of benchmarks exist, which are based on the google benchmark library, and can be run from command-line. For this, the ```-DDETRAY_BUILD_BENCHMARKS=ON``` and ``` -DDETRAY_BUILD_CLI_TOOLS=ON``` flags need to be specified. Then pass the detray detector file(s) and additional options to the benchmark tools for the different hardware backends:
```

Comment:
Nit: These code blocks should be single ticks, also there is an additional space in the second one.

Reply:
fixed